typings/enums.d.ts

```diff
@@ -152,3 +152,140 @@ export enum WebhookTypes {
   Incoming = 1,
   'Channel Follower' = 2,
 }
+
+export enum ConstantsEvents {
+  RATE_LIMIT = 'rateLimit',
+  INVALID_REQUEST_WARNING = 'invalidRequestWarning',
+  CLIENT_READY = 'ready',
+  APPLICATION_COMMAND_CREATE = 'applicationCommandCreate',
+  APPLICATION_COMMAND_DELETE = 'applicationCommandDelete',
+  APPLICATION_COMMAND_UPDATE = 'applicationCommandUpdate',
+  GUILD_CREATE = 'guildCreate',
+  GUILD_DELETE = 'guildDelete',
+  GUILD_UPDATE = 'guildUpdate',
+  INVITE_CREATE = 'inviteCreate',
+  INVITE_DELETE = 'inviteDelete',
+  GUILD_UNAVAILABLE = 'guildUnavailable',
+  GUILD_MEMBER_ADD = 'guildMemberAdd',
+  GUILD_MEMBER_REMOVE = 'guildMemberRemove',
+  GUILD_MEMBER_UPDATE = 'guildMemberUpdate',
+  GUILD_MEMBER_AVAILABLE = 'guildMemberAvailable',
+  GUILD_MEMBERS_CHUNK = 'guildMembersChunk',
+  GUILD_INTEGRATIONS_UPDATE = 'guildIntegrationsUpdate',
+  GUILD_ROLE_CREATE = 'roleCreate',
+  GUILD_ROLE_DELETE = 'roleDelete',
+  GUILD_ROLE_UPDATE = 'roleUpdate',
+  GUILD_EMOJI_CREATE = 'emojiCreate',
+  GUILD_EMOJI_DELETE = 'emojiDelete',
+  GUILD_EMOJI_UPDATE = 'emojiUpdate',
+  GUILD_BAN_ADD = 'guildBanAdd',
+  GUILD_BAN_REMOVE = 'guildBanRemove',
+  CHANNEL_CREATE = 'channelCreate',
+  CHANNEL_DELETE = 'channelDelete',
+  CHANNEL_UPDATE = 'channelUpdate',
+  CHANNEL_PINS_UPDATE = 'channelPinsUpdate',
+  MESSAGE_CREATE = 'messageCreate',
+  MESSAGE_DELETE = 'messageDelete',
+  MESSAGE_UPDATE = 'messageUpdate',
+  MESSAGE_BULK_DELETE = 'messageDeleteBulk',
+  MESSAGE_REACTION_ADD = 'messageReactionAdd',
+  MESSAGE_REACTION_REMOVE = 'messageReactionRemove',
+  MESSAGE_REACTION_REMOVE_ALL = 'messageReactionRemoveAll',
+  MESSAGE_REACTION_REMOVE_EMOJI = 'messageReactionRemoveEmoji',
+  THREAD_CREATE = 'threadCreate',
+  THREAD_DELETE = 'threadDelete',
+  THREAD_UPDATE = 'threadUpdate',
+  THREAD_LIST_SYNC = 'threadListSync',
+  THREAD_MEMBER_UPDATE = 'threadMemberUpdate',
+  THREAD_MEMBERS_UPDATE = 'threadMembersUpdate',
+  USER_UPDATE = 'userUpdate',
+  PRESENCE_UPDATE = 'presenceUpdate',
+  VOICE_SERVER_UPDATE = 'voiceServerUpdate',
+  VOICE_STATE_UPDATE = 'voiceStateUpdate',
+  TYPING_START = 'typingStart',
+  WEBHOOKS_UPDATE = 'webhookUpdate',
+  INTERACTION_CREATE = 'interactionCreate',
+  ERROR = 'error',
+  WARN = 'warn',
+  DEBUG = 'debug',
+  SHARD_DISCONNECT = 'shardDisconnect',
+  SHARD_ERROR = 'shardError',
+  SHARD_RECONNECTING = 'shardReconnecting',
+  SHARD_READY = 'shardReady',
+  SHARD_RESUME = 'shardResume',
+  INVALIDATED = 'invalidated',
+  RAW = 'raw',
+  STAGE_INSTANCE_CREATE = 'stageInstanceCreate',
+  STAGE_INSTANCE_UPDATE = 'stageInstanceUpdate',
+  STAGE_INSTANCE_DELETE = 'stageInstanceDelete',
+}
+
+export enum ConstantsShardEvents {
+  CLOSE = 'close',
+  DESTROYED = 'destroyed',
+  INVALID_SESSION = 'invalidSession',
+  READY = 'ready',
+  RESUMED = 'resumed',
+}
+
+export enum ConstantsColors {
```

Comment:
This is not an enum, but TS allows compiling something like `Colors[Colors.RED]`, yielding a `string`, however in runtime it's actually `undefined`.
We could make it an enum however, `createEnum` could be modified to accept objects too.